### PR TITLE
Introduce RemoteTableName class to MongoDB

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Introduce `RemoteTableName` class to MongoDB

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
@hashhar @Praveen2112 Could you review this PR when you have time? This partially relates to FTE PR #15062

File: plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoTableHandle.java [hunk 1]
```diff
@@ -40,7 +42,9 @@ public void testRoundTripWithoutQuery()
     @Test
     public void testRoundTripWithQuery()
     {
-        MongoTableHandle expected = new MongoTableHandle(new SchemaTableName("schema", "table"), Optional.of(new Document("key", "value")));
+        SchemaTableName schemaTableName = new SchemaTableName("schema", "table");
+        RemoteTableName remoteTableName = new RemoteTableName("schema", "table");
```
Comment (Praveen2112) [inline plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoTableHandle.java:46]: nit: Can we add a test case with mixed case names for `remoteTableName`

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/RemoteTableName.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+public final class RemoteTableName
+{
+    private final String databaseName;
+    private final String collectionName;
+
+    @JsonCreator
+    public RemoteTableName(String databaseName, String collectionName)
```
Comment (Praveen2112) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/RemoteTableName.java:29]: Should we add `@JsonProperty` ?

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java [hunk 3]
```diff
@@ -166,7 +166,7 @@ public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSess
         ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
         for (SchemaTableName tableName : listTables(session, prefix)) {
             try {
-                columns.put(tableName, getTableMetadata(session, tableName).getColumns());
+                columns.put(tableName, getTableMetadata(tableName).getColumns());
```
Comment (Praveen2112) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:169]: nit: Can we extract this as a different commit ?

**Review by Praveen2112 — APPROVED** [review]
% changes. LGTM
